src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,355 @@
+from typing import Dict, List, Mapping, Optional, Sequence
+
+from tiledb.cloud import dag
+from tiledb.cloud.utilities import as_batch
+
+
+def ingest_files(
+    file_dir_uri: str,
+    index_uri: str,
+    file_name: Optional[str] = None,
+    acn: Optional[str] = None,
+    config=None,
+    environment_variables: Optional[Mapping[str, str]] = None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+    trace_id: Optional[str] = None,
+    # Index creation params
+    index_type: str = "IVF_FLAT",
+    index_creation_kwargs: Optional[Dict] = None,
+    # DirectoryTextReader params
+    include: str = "*",
+    exclude: Optional[Sequence[str]] = ("[.]*", "*/[.]*"),
+    suffixes: Optional[Sequence[str]] = None,
+    max_files: Optional[int] = None,
+    text_splitter: str = "RecursiveCharacterTextSplitter",
+    text_splitter_kwargs: Optional[Dict] = None,
+    # Embedding params
+    embedding_class: str = "LangChainEmbedding",
+    embedding_kwargs: Optional[Dict] = None,
+    openai_key: Optional[str] = None,
+    # Index update params
+    index_timestamp: Optional[int] = None,
+    workers: int = -1,
+    worker_resources: Optional[Dict] = None,
+    worker_image: Optional[str] = None,
+    extra_worker_modules: Optional[List[str]] = None,
+    driver_resources: Optional[Dict] = None,
+    driver_image: Optional[str] = None,
+    extra_driver_modules: Optional[List[str]] = None,
+    max_tasks_per_stage: int = -1,
+    embeddings_generation_mode: dag.Mode = dag.Mode.LOCAL,
```

Comment:
Everything must default to batch mode. Running this in local is unexpected. The goals are that like all other verticals we support and default to batch ingestion capabilities.

Reply:
Document indexing has multiple execution steps that can be run in different modes:
- `ingest_files` creates a `BATCH` taskgraph that runs all the indexing. This means that all processing happens within a `BATCH` taskgraph with `access_credentials` even if the options here are set to `LOCAL`
- `embeddings_generation`: reads the documents and creates text embeddings. This can spawn its own taskgraph.
- `vector_indexing`: creates a vector index from the produced embeddings. This can spawn its own taskgraph.

The default configuration at the moment is:
- `ingest_files` creates a `BATCH` taskgraph that runs all the indexing.
- `embeddings_generation, vector_indexing` run in `LOCAL` mode within a UDF of the `ingest_files` taskgraph. Both of these tasks can leverage the available parallelism within the single worker. 

This is expected to be a good default execution configuration for cost and latency even for sets of thousands of documents. 

Do you want all the execution steps to be executed in `BATCH` mode by default?

Reply:
The requirement again, as we discussed and as spelled out in the story is to have a robust batch mode ingestion that can scale to millions of documents. Local mode is a bad default and does not meet our intended goals.